### ISSUE TITLE
Implement the new WhatsApp encryption system

### ISSUE THREAD

**Issue by VictorOsorioMedrano** [mozillahispano/openwapp#25]
WhatsApp now includes an encryption system, this is a important point for implement on OpenWapp

Here the news: http://www.genbeta.com/mensajeria-instantanea/whatsapp-da-un-salto-enorme-en-seguridad-integrando-cifrado-de-punto-a-punto

**Comment by timothymctim** [comment]
It seems nobody has started to work on this.  Am I right?

**Comment by duesenfranz** [comment]
@timothymctim 
I started, but didn't upload anything so far...
I think I will push this weekend... if you want to help, I can try earlier :)

**Comment by timothymctim** [comment]
@duesenfranz thank you for your reply.  It's good to know someone is still working on it :smiley:
I'm not sure whether I could help, since I don't know anything of the WA protocol, nor the CoSeMe code, moreover I can only spend limited time on this.  But, if you have a specific problem, I could dive into it.

**Comment by Timmaks** [comment]
WhatsAPI committed a workaround : https://github.com/mgp25/WhatsAPI-Official/commit/4576f4c9df7cc0f36053be94524ae5261490d22f

**Comment by Timmaks** [comment]
I think we can close this issue because at the moment it is not a issue anymore

**Comment by duesenfranz** [comment]
Yeah sorry, I was *very* busy the last weeks. I did a lot of work before, but turns out the guys from TextSecure meanwhile also worked [on moving their axolotl stuff into an own library](https://github.com/WhisperSystems/TextSecure-Browser/commits/libaxolotl). As far as I can see, this is still an ongoing effort. So I agree the best solution is closing this for now. The longer we can wait, the better, because then we may be able to simply plug their library in.

Maybe they even put it in their own repository... :sunny:

Sorry again for not answering so long :)
